proposals/vector_to_points.json

```
@@ -0,0 +1,68 @@
+{
+    "id": "vector_to_points",
+    "summary": "Sample random points from geometries",
+    "description": "For each geometry samples a number of random points based on one of the sampling method.",
+    "categories": [
+        "cubes",
+        "vector"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "Input geometries for sample extraction.\n\nTo maximize interoperability, a nested `GeometryCollection` should be avoided. Furthermore, a `GeometryCollection` composed of a single type of geometries should be avoided in favour of the corresponding multi-part type (e.g. `MultiPolygon`).",
+            "schema": [
+                {
+                    "type": "object",
+                    "subtype": "geojson"
+                },
+                {
+                    "type": "object",
+                    "subtype": "vector-cube"
+                }
+            ]
+        },
+        {
+            "name": "count",
+            "description": "The number of points to compute. For each point given the maximum number of computed points is always 1.",
```

Bit unclear if this parameter is total number  or number per geometry.

I assume it must be the former (total number). But other parts of description seem to suggest the latter:

> For each geometry samples a number of random points

> For each point given ... number is always 1

---

> For each point given the maximum number of computed points is always 1.

This sentence was very confusing to me. I guess it describes the behavior when your input geometries contain points. I think that discussing that edge case here makes things more confusing than it should be.

---

Hmm, I guess I'd check what implementations do to not make the implementation overly complicated. 

```suggestion
            "description": "The number of points to compute. Does not apply for point geometries.",
```

---

I think part of my confusion (also see my first comment in this side-thread) is that I first assumed the input data is a geojson object. I somehow missed the part that the input data can also be a vector cube. 

So `count` is the number of samples per "feature" (in geojson sense)?

---

Yes, I think that was the intention, but as said, let's check implementations, too.

---

> Does not apply for point geometries for which the count is always `1`.

And what about MultiPoints or geometry collections with multiple points? You could cover that generically with something like:

> Note that points in the input geometries can be selected at most once by the sampling.

---

That sounds pretty good, indeed! 👍

```suggestion
            "description": "The number of points to compute per geometry. Note that points in the input geometries can be selected at most once by the sampling.",
```